tests/integ/sagemaker/workflow/test_selective_execution.py

```diff
@@ -0,0 +1,201 @@
+        """adds two numbers"""
+        return a + b
+
+    step_output_a = generator()
+    step_output_b = sum(step_output_a[0], step_output_a[1])
+
+    pipeline = Pipeline(
+        name=pipeline_name,
+        steps=[step_output_b],
+        sagemaker_session=sagemaker_session,
+    )
+
+    try:
+        execution, _ = create_and_execute_pipeline(
+            pipeline=pipeline,
+            pipeline_name=pipeline_name,
+            region_name=region_name,
+            role=role,
+            no_of_steps=2,
+            last_step_name="sum",
+            execution_parameters=dict(),
+            step_status="Succeeded",
+            step_result_type=int,
+            step_result_value=7,
+        )
+
+        create_and_execute_pipeline(
+            pipeline=pipeline,
+            pipeline_name=pipeline_name,
+            region_name=region_name,
+            role=role,
+            no_of_steps=2,
+            last_step_name="sum",
+            execution_parameters=dict(),
+            step_status="Succeeded",
+            step_result_type=int,
+            step_result_value=7,
+            selective_execution_config=SelectiveExecutionConfig(
+                source_pipeline_execution_arn=execution.arn,
+                selected_steps=[get_step(step_output_b).name],
+            ),
+        )
+
+    finally:
+        try:
+            pipeline.delete()
+        except Exception:
+            pass
+
+
+def test_selective_execution_of_regular_step_depended_by_function_step(
+    sagemaker_session,
+    role,
+    pipeline_name,
+    region_name,
+    dummy_container_without_error,
+    sklearn_latest_version,
+):
+    # Test Selective Pipeline Execution on regular step -> [select: function step]
+    os.environ["AWS_DEFAULT_REGION"] = region_name
+
+    script_path = os.path.join(DATA_DIR, "dummy_script.py")
+
+    sklearn_processor = SKLearnProcessor(
+        framework_version=sklearn_latest_version,
+        role=role,
+        instance_type=INSTANCE_TYPE,
+        instance_count=1,
+        command=["python3"],
+        sagemaker_session=sagemaker_session,
+        base_job_name="test-sklearn",
+    )
+
+    step_sklearn = ProcessingStep(
+        name="sklearn-process",
+        processor=sklearn_processor,
+        code=script_path,
+    )
+
+    @step(
+        role=role,
+        instance_type=INSTANCE_TYPE,
+        image_uri=dummy_container_without_error,
+        keep_alive_period_in_seconds=60,
+    )
+    def func_2(arg):
+        return arg
+
+    final_output = func_2(step_sklearn.properties.ProcessingJobStatus)
+
+    pipeline = Pipeline(
+        name=pipeline_name,
+        steps=[final_output],
+        sagemaker_session=sagemaker_session,
+    )
+
+    try:
+        execution, _ = create_and_execute_pipeline(
+            pipeline=pipeline,
+            pipeline_name=pipeline_name,
+            region_name=region_name,
+            role=role,
+            no_of_steps=2,
+            last_step_name="func",
```

Comment:
Am I missing something or should this be func_2 ?

Reply:
Yea, this part looks weird. I copied the test from an existing test and made updates based on that. Will double check and correct them as needed

Reply:
Checked the test helper logic, it simply verifies `assert last_step_name in execution_steps[0]["StepName"]`, as the final step's name starts with `func_2`, it would not break the assert statement.

I can update the var to be `last_step_name_prefix` in next change if no further critical updates required in this PR.